src/compas_rhino/geometry/trimesh/curvature.py

```diff
@@ -0,0 +1,331 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import Rhino
+from Rhino.Geometry.Vector3d import Multiply, CrossProduct
+from Rhino.Geometry.Point3d import FromPoint3f
+
+from math import pi
+from math import sqrt
+from math import atan2
+
+from clr import StrongBox
+from scriptcontext import doc
+from System import Array
+
+from compas.plugins import plugin
+
+
+__all__ = [
+    'trimesh_gaussian_curvature',
+    'trimesh_mean_curvature',
+    'trimesh_principal_curvature',
+]
+
+
+@pluggable(category="trimesh", requires=['Rhino'])
+def trimesh_gaussian_curvature(M):
+    """Compute the discrete Gaussian curvature of a triangle mesh.
+
+    Parameters
+    ----------
+    M : (list, list)
+        A mesh represented by a list of vertices and a list of faces.
+
+    Returns
+    -------
+    list
+        The discrete Gaussian curvature per vertex.
+
+    Notes
+    -----
+    Description: The angle defect at a vertex is used to describe the Gaussian curvature in a neighborhood around a vertex.
+
+    Notation Convention:
+    :math:`K` - discrete Gaussian curvature
+    :math:`j,k` - the vertices from the Star of vertex i
+    :math:`e_{ij}, e_{ik}` - the vector from vertex i to j, i to k
+    :math:`\theta_{i}^{jk}` - interior angle at vertex i of triangle ijk
+
+    Formula:
+    .. math::
+
+        K_{i} = 2\pi-\sum\theta_{i}^{jk}
+
+    Examples
+    --------
+    Make a mesh from scratch
+    >>> from compas.geometry import Sphere
+    >>> sphere = Sphere([1, 1, 1], 1)
+    >>> sphere = Mesh.from_shape(sphere, u=30, v=30)
+    >>> sphere.quads_to_triangles()
+    >>> M = sphere.to_vertices_and_faces()
+
+    Compute the discrete Gaussian curvature
+
+    >>> K = trimesh_gaussian_curvature(M)
+
+    References
+    ----------
+    .. [1] Formula of Discrete Gaussian Curvature available at Keenan Crane's lecture, 03:16-07:11, at https://youtu.be/sokeN5VxBB8
+
+    """
+    # (1) see if input is already Rhino.Geometry.Mesh
+    mesh = Rhino.Geometry.Mesh()
+    if type(M) != Rhino.Geometry.Mesh:
```

Comment:
```suggestion
    if not isinstance(M, Rhino.Geometry.Mesh):
```
Sometimes `type` can give unexpected results...